Sources/FunctionalTableData/CellStyle.swift

```diff
@@ -159,6 +163,10 @@ public struct CellStyle {
 			cell.selectedBackgroundView = nil
 		}
 		
+		if let selectedBackgroundView = selectionBackgroundViewProvider?.backgroundView() {
+			cell.selectedBackgroundView = selectedBackgroundView
+		}
```

Comment:
Shouldn’t this code complete the color one? I should be able to set a custom view and color right? Either case, you should also undo anything you do in an else statement

Reply:
I wasn't sure if this entire solution is the correct way to go yet. 
Setting both a `selectedBackgroundColor` and `selectionColor` seems to override the `selectionColor`, similar to what happens if you provide both a `backgroundColor` and a `backgroundViewProvider`. 
🤔  Do you think it's more straightforward if we only respect the `selectedBackgroundViewProvider` if a `selectionColor` is not provided?

Reply:
Ah, so providing both a `backgroundColor` and a `backgroundSelectionViewProvider` sets the background colour and also the cell's `backgroundView`. In some circumstances, you won't see the resulting `backgroundColor`. 

But providing both a `selectionColor` and a `selectionBackgroundViewProvider` is different.. so maybe it should take an enum that can be either a `selectionColor` or a `selectionBackgroundViewProvider`, but not both.

Reply:
> Do you think it's more straightforward if we only respect the selectedBackgroundViewProvider if a selectionColor is not provided?

I would check what iOS is doing in this situation, what happens when you set a background view with a selection color?

But, as for my comment above `if let selectedBackgroundView = selectionBackgroundViewProvider?.backgroundView() {` the else statement of this should probably be removing the view, but that is going to clash with the previous selectionColor code. 

You might want to add tests to this new situation making sure the background and colors have the expected values depending on the configuration (color + background, color only, background only, etc)

Reply:
Setting the `selectionColor` overrides the `backgroundColor` for selected cells. 

Maybe it makes sense to do something like...
```
cell.selectedBackgroundView = nil
if let selectionColor = selectionColor {
	let selectedBackgroundView = UIView()
	selectedBackgroundView.backgroundColor = selectionColor
	cell.selectedBackgroundView = selectedBackgroundView
}
		
if let selectedBackgroundView = selectionBackgroundViewProvider?.backgroundView() {
	cell.selectedBackgroundView = selectedBackgroundView
}
```